batch-l2.md

```diff
@@ -0,0 +1,1494 @@
+  environments?: Array<{ [key:string]: string }>
+  executionRoleArn?: iam.IRole
+  fargateVersion?: FargateVersion
+  image: ContainerImage
+  jobRoleArn?: iam.IRole
+  linuxParameters?: LinuxParameters
+  logDriver?: LogDriver
+  // Memory is deprecated in favor of ResourceRequirements
+  // mountpoints added via addMountPoint() or addMountVolume()
+  disableNetworking?: boolean
+  priveleged?: boolean
+  readonlyRootFileSystem?: boolean
+  // image covers ContainerProperties.ResourceRequirements.MEMORY
+  cpu: number
+  gpuCount?: number
+  secrets?: { [key: string]: Secret }
+  // Ulimits are added via addUlimit()
+  user?: string
+  // Vcpus is deprecated in favor of ResourceRequirements
+  // Volumes are added via addVolume() or addMountedVolume()
+
+  addUlimit(...Ulimit[])
+  addVolume(...EcsVolume[])
+  addMountedVolume(..MountedEcsVolume[])
+}
+
+enum FargateVersion {
+// TODO
+
+}
+
+interface Ulimit {
+  hardLimit: number;
+  name: UlimitName;
+  softLimit: number;
+}
+
+interface UlimitName {
+  //TODO
+}
+
+interface MountedEcsVolume {
+  name: string;
+  efsVolumeConfiguration?: EfsVolumeConfiguration;
+  host?: string;
+  containerPath: string;
+  readOnly: boolean;
+}
+
+interface EfsVolumeConfiguration {
+  fileSystemId: string;
+  rootDirectory?: string;
+  transitEncryption?: string;
+  transitEncryptionPort?: number;
+  authorizationConfig?: AuthorizationConfig;
+}
+
+interface AuthorizationConfig {
+  accessPointId?: string;
+  iam?: string;
+}
+
+interface EcsJobDefinitionProps {
+  containerDefinition: EcsContainerDefinition
+  fargatePlatformVersion?: FargatePlatformVersion
+  compatibility?: Compatibility
+}
+
+class EcsJobDefinition implements IEcsJobDefinition {
+  constructor(readonly props: EcsJobDefinitionProps = {}) {}
+}
+```
+
+* `IEksJobDefinition`
+
+```ts
+interface IEksJobDefinition extends IJobDefinition {
+  pod: batch.IEksPod;
+  platform?: Platform;
+}
+
+enum Platform {
+  // TODO
+}
+
+class EksPod {
+  containers: batch.EksContainerDefinition[]
+  dnsPolicy?: DnsPolicy
+  useHostNetwork?: boolean
+  serviceAccount?: string
+
+  addContainer(batch.EksContainerDefinition)
+}
+
+class EksContainerDefinition {
+  image: string
+  args?: string[]
+  command?: string[]
+  env?: { [key:string]: string }
+  imagePullPolicy?: ImagePullPolicy
+  name?: string
+  resources?: EksContainerResources
+  priveleged?: boolean
+  readonlyFileSystem?: boolean
+  runAsGroup?: number
+  runAsRoot?: boolean
+  runAsUser?: number
+  volumes?: EksVolume[]
+
+  addVolume(...EksVolume[])
+}
+
+// TODO: need props for all of these smaller classes
+// TODO: EksContainerResources has no definition
+abstract class EksVolume {
+  name: string;
+  mountPath?: string;
+  readonly?: boolean;
+}
+
+class EmptyDirVolume extends EksVolume {
+  medium?: MediumType;
+  sizeLimit?: number;
+}
+
+class HostPathVolume extends EksVolume {
+  path: string;
+}
+
+class SecretPathVolume extends EksVolume {
+  secret: ssm.Secret;
```

Comment:
Kube (eks) secret is just a volume location pointing a a kube secret resource.  SSM is not directly involved in this mapping.

customers have the ability to populate kube secrets from SSM, but those are still exposed to pods as a generic volume location and secretName string.